src/core/public/overlays/sidecar/components/resizable_button.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useRef } from 'react';
+import classNames from 'classnames';
+import './resizable_button.scss';
+import { getPosition } from '../helper';
+import { ISidecarConfig, SIDECAR_DOCKED_MODE } from '../sidecar_service';
+
+interface Props {
+  onResize: (size: number) => void;
+  flyoutSize: number;
+  dockedMode: ISidecarConfig['dockedMode'] | undefined;
+}
+
+export const MIN_SIDECAR_SIZE = 350;
+
+export const ResizableButton = ({ dockedMode, onResize, flyoutSize }: Props) => {
+  const isHorizontal = dockedMode !== SIDECAR_DOCKED_MODE.TAKEOVER;
+
+  const classes = classNames('resizableButton', {
+    'resizableButton--vertical': !isHorizontal,
+    'resizableButton--horizontal': isHorizontal,
+  });
+
+  const initialMouseXorY = useRef(0);
+  const initialFlyoutSize = useRef(flyoutSize);
```

Comment:
How about add `initialFlyoutSize.current = flyoutSize` here. Then the `onMouseDown` doesn't need to depends on flyoutSize.

Reply:
We couldn't, this will cause calculation error. And I think we should keep `initialFlyoutSize` assigning value only in `onMouseMove` to make it clear although there will be a more depence.

Reply:
You're right. Need to add one more useRef to reduce efforts. The code may like
```ts
const flyoutSizeRef = useRef(flyoutSize);
flyoutSizeRef.current = flyoutSize;
const initialFlyoutSize = useRef(flyoutSize);

initialFlyoutSize.current = flyoutSizeRef.current;
```
Current implementation is good to me. It's a little bit complicated.

Reply:
Yes, this way needs to add one more useRef.